src/lib.rs

```diff
@@ -0,0 +1,1036 @@
+        Operand::new(
+            vreg,
+            OperandPolicy::Reg,
+            OperandKind::Def,
+            OperandPos::Before,
+        )
+    }
+    #[inline(always)]
+    pub fn reg_reuse_def(vreg: VReg, idx: usize) -> Self {
+        Operand::new(
+            vreg,
+            OperandPolicy::Reuse(idx),
+            OperandKind::Def,
+            OperandPos::After,
+        )
+    }
+    #[inline(always)]
+    pub fn reg_fixed_use(vreg: VReg, preg: PReg) -> Self {
+        Operand::new(
+            vreg,
+            OperandPolicy::FixedReg(preg),
+            OperandKind::Use,
+            OperandPos::Before,
+        )
+    }
+    #[inline(always)]
+    pub fn reg_fixed_def(vreg: VReg, preg: PReg) -> Self {
+        Operand::new(
+            vreg,
+            OperandPolicy::FixedReg(preg),
+            OperandKind::Def,
+            OperandPos::After,
+        )
+    }
+
+    #[inline(always)]
+    pub fn vreg(self) -> VReg {
+        let vreg_idx = ((self.bits as usize) & VReg::MAX) as usize;
+        VReg::new(vreg_idx, self.class())
+    }
+
+    #[inline(always)]
+    pub fn class(self) -> RegClass {
+        let class_field = (self.bits >> 25) & 1;
+        match class_field {
+            0 => RegClass::Int,
+            1 => RegClass::Float,
+            _ => unreachable!(),
+        }
+    }
+
+    #[inline(always)]
+    pub fn kind(self) -> OperandKind {
+        let kind_field = (self.bits >> 27) & 3;
+        match kind_field {
+            0 => OperandKind::Def,
+            1 => OperandKind::Mod,
+            2 => OperandKind::Use,
+            _ => unreachable!(),
+        }
+    }
+
+    #[inline(always)]
+    pub fn pos(self) -> OperandPos {
+        let pos_field = (self.bits >> 26) & 1;
+        match pos_field {
+            0 => OperandPos::Before,
+            1 => OperandPos::After,
+            _ => unreachable!(),
+        }
+    }
+
+    #[inline(always)]
+    pub fn policy(self) -> OperandPolicy {
+        let policy_field = (self.bits >> 29) & 7;
+        let preg_field = ((self.bits >> 20) as usize) & PReg::MAX;
+        match policy_field {
+            0 => OperandPolicy::Any,
+            1 => OperandPolicy::Reg,
+            2 => OperandPolicy::Stack,
+            3 => OperandPolicy::FixedReg(PReg::new(preg_field, self.class())),
+            4 => OperandPolicy::Reuse(preg_field),
+            _ => unreachable!(),
+        }
+    }
+
+    #[inline(always)]
+    pub fn bits(self) -> u32 {
+        self.bits
+    }
+
+    #[inline(always)]
+    pub fn from_bits(bits: u32) -> Self {
+        debug_assert!(bits >> 29 <= 4);
+        Operand { bits }
+    }
+}
+
+impl std::fmt::Debug for Operand {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        std::fmt::Display::fmt(self, f)
+    }
+}
+
+impl std::fmt::Display for Operand {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(
+            f,
+            "{:?}@{:?}: {}{} {}",
```

Comment:
`Def@After` and `Use@Before` are by far the most common cases. I find it more readable to shorten these to just `Def` and `Use` (and keep using the long form for `Def@Before` and `Use@After`).

Reply:
Personally I found the consistent `@Before` / `@After` suffixes helpful near the end of long debugging sessions, the sort where my eyes start to glaze over with gigabyte-sized log files, and the `Use` -> `Before` / `Def` -> `After` mapping would have been just a bit more cognitive load. But it's possible that my use-case (or my brain) is weird, and I'm happy to change this if you insist :-)

Reply:
Well, it's a combination of two factors:
1. I never actually use `Use@After` or `Def@Before`, so there's no confusion on my end. I suspect that in a real world context (i.e. not fuzzing) you will end up with almost no uses of `Use@After`/`Def@Before` and would want those to stand out in the rare cases where they are used.
2. I have instructions with a *lot* of operands which results in extremely long lines. In these lines the `@After`/`@Before` are just noise which don't add any information:
```
exit_indirect.ret exit55 Use@Before: v595i fixed(p28i), Use@Before: v567i fixed(p8i), Use@Before: v568i fixed(p9i), Use@Before: v569i fixed(p10i), Use@Before: v570i fixed(p11i), Use@Before: v587i fixed(p12i), Use@Before: v588i fixed(p13i), Use@Before: v589i fixed(p14i), Use@Before: v590i fixed(p15i), Use@Before: v591i fixed(p16i), Use@Before: v592i fixed(p17i), Use@Before: v593i fixed(p18i), Use@Before: v594i fixed(p19i), Use@Before: v579i fixed(p20i), Use@Before: v586i fixed(p2i), Use@Before: v581i fixed(p21i), Use@Before: v582i fixed(p22i), Use@Before: v583i fixed(p23i), Use@Before: v584i fixed(p24i), Use@Before: v580i fixed(p25i), Use@Before: v22f fixed(p0f), Use@Before: v23f fixed(p1f), Use@Before: v24i stack, Use@Before: v25i stack, Use@Before: v26i stack, Use@Before: v27i stack, Use@Before: v28f stack, Use@Before: v29f stack, Use@Before: v30f stack, Use@Before: v31f stack, Use@Before: v32f stack, Use@Before: v33f stack, Use@Before: v34f stack, Use@Before: v35f stack, Use@Before: v585f stack, Use@Before: v37f stack, Use@Before: v38f stack, Use@Before: v39f stack, Use@Before: v40f stack, Use@Before: v41f stack, Use@Before: v42f stack, Use@Before: v43f stack, Use@Before: v572f stack, Use@Before: v45f stack, Use@Before: v46f stack, Use@Before: v47f stack, Use@Before: v48f stack, Use@Before: v49f stack, Use@Before: v50f stack, Use@Before: v51f stack, Use@Before: v52f stack, Use@Before: v53f stack, Use@Before: v54f stack, Use@Before: v55f stack, Use@Before: v56f stack, Use@Before: v57f stack, Use@Before: v58f stack, Use@Before: v59f stack, Use@Before: v60i fixed(p26i), Use@Before: v61f fixed(p2f), Use@Before: v62f fixed(p3f)
```

Reply:
Fair enough -- I think that (i) now that I've hopefully passed the worst of the RA2-debugging, and (ii) this will mostly show up in others' logs now, I agree that brevity is a reasonable principle :-)

FWIW, at least in Cranelift, `Def@Before` has been used for temps, so it is *somewhat* common, though it's still the odd case out.